features/step_definitions/lesson_steps.rb

```diff
@@ -0,0 +1,75 @@
+Given /a user named '([^']+)' exists$/ do |user_name|
+  FactoryGirl.create(:user, username: user_name)
+end
+
+Given /^a course named '([^']+)' exists$/ do |course_name|
+  @course = FactoryGirl.create(:course, title: course_name)
+end
+
+Given /^a section named '([^']+)' exists$/ do |section_name|
+  @section = FactoryGirl.create(:section, title: section_name, course: @course)
+end
+
+Given /^the following lessons exist:$/ do |table|
+  table.hashes.each do |hash|
+    FactoryGirl.create(:lesson, title: hash[:lesson_name], section: @section)
+  end
+end
+
+Given /^I am logged in$/ do
+  @user = FactoryGirl.create(:user)
+  log_in @user
+end
+
+Given /I go to the '([^']+)' course/ do |course_name|
+  course_link_text = "#{Course.count}: #{course_name}"
+  click_link course_link_text
+  expect(page).to have_content(@section.title)
+end
+
+Given /^no lessons are completed$/ do 
+  within ".lc-percent-completion" do
+    expect(page).to have_content('0% Completed')
```

Comment:
Would it be better to do this at the db level instead? something like `expect(@user.completed_lessons.count).to eql(0)`